### DISCUSSION TITLE
Empty `retype.js`

### DISCUSSION THREAD

**Discussion by DannyBen** [retypeapp/retype#88] [Q&A] [answered]
Something weird is happening.

I have a simple two page test retype site, that worked 30 minutes ago.
But now, I see that nothing is loaded in the sidebar, and all javascript functions are off (search, theme...).

Digging deeper, I see that the `retype.js` file in the output folder is just blank.

Reinstalling retype didn't help.

Is this a bug? How did I break it?

**Comment by geoffreymcgill** [comment, marked as answer]
I don't think we have run into this issue yet, so it's new to me. 

By default, Retype will emit the website to a `.retype` folder. Can you delete that folder, and then try `retype watch` again? First shutdown the `retype watch` server using `alt + c`.

If that doesn't help, shutdown again, delete the `.retype` folder again, and then run the command `retype watch --verbose`, which will turn on verbose logging. If you can, please share that console output with us here and we might be able to see something in the generated verbose log.

**Reply by DannyBen** [reply]
> I don't think we have run into this issue yet, so it's new to me.

Yeah - I figured it is something broken on my system, and not a retype bug - this is why I posted here and not as an issue.

More weirdness:

- The result is the same after deleting the folder, and even after rebooting my VM.
- This works when I do it in another folder.
- This used to work in the exact same folder where it is now broken.

I will recreate this folder, see if it helps - maybe something got stuck in the OS.

I will post here if I solve this - just thought someone might have seen this before.

**Reply by DannyBen** [reply]
Recreating the folder completely seemed to have solved it - was probably something in my file system. Thanks for helping, I am closing this.

**Reply by fabriciomurta** [reply]
This looks like a file locking issue. for some reason retype.js could have been locked by something that didn't allow it to be rewritten ... but again, linux file systems/kernel allows to "unlink" an open file so that whatever is locking it still get its old contents while new queries get the new file.... but as you even rebooted the vm and the lock was still in... maybe you using filesystem shared with the VM's host (so that the host could be the one locking up the file, and being Windows, it'd make a lot of sense of the file being b0rked as that). But yet again, if it was windows locking issue, you wouldn't have been able to simply wipe the directory.

It really looks like an OS issue, as if a antispyware somehow locked the file thinking it has malicious code...

**Reply by DannyBen** [reply]
Yes. This never happened to me since - and I rebuilt Retype sites for the past 48 hours.

It was probably not a locking issue, as you mentioned - I was fully able to delete it, and then rebuilding brought an empty file again. 

I think it happened when I - by mistake - set up my system to watch both the source and target directories, so it found itself (without me noticing) in an endless loop of rebuilding - so perhaps an inodes related issue.

(I am using a custom watch, due to https://github.com/retypeapp/retype/issues/86).

**Reply by geoffreymcgill** [reply]
If you ever come across this again and can figure out how to reliably reproduce, please let us know. We will try and reproduce the conditions and see if there's anything Retype can do to prevent from happening.
